### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by achadha**

### CODE REVIEW

**Comment by jfhamlin** [comment]
A few high-level questions and comments:

1. Let's squash these into a couple of commits before merging!
2. Is there a test harness?
3. What do we need to do to get the CI build working?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Include your project-specific ignores in this file
+# Read about how to use .gitignore: https://help.github.com/articles/ignoring-files
+
+build
+node_modules
+ncp-debug.log
+npm-debug.log
+restructure/dist
+.DS_Store
```
Comment (jfhamlin) [inline .gitignore]: This isn't really a project-specific file. We should all have this in a global .gitignore: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Comment by achadha** [comment]
2. At the moment, no. Need to do that.
3. I think we need to add 'sudo: required' to the travis.yml. I feel like it shouldn't be necessary, but something is up with the permissions, and that was my way around it.

File: components/Resource/Resource.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+class Resource extends React.Component {
+	constructor() {
+		super();
+		this.state = {resource: {}};
+	}
+
+	loadResourceFromServer() {
+		let { query } = this.props.location;
+		let resourceID = query.id;
+		let url = 'http://localhost:3000/resources/' + resourceID;
```
Comment (jfhamlin) [inline components/Resource/Resource.js]: Can we make the API URL configurable, and initialize it in one place?

**Pull request merged**